client/state/audio/middleware.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Internal dependencies
+ */
+import {
+	HAPPYCHAT_RECEIVE_EVENT,
+} from 'state/action-types';
+
+const isAudioSupported = () => typeof window === 'object' && typeof window.Audio === 'function';
+
+export const playSound = src => {
+	if ( ! isAudioSupported() ) {
+		return;
+	}
+
+	const audioClip = new window.Audio( src );
+	audioClip.play();
+};
+
+export const onHappyChatMessage = ( dispatch, { event } ) => {
+	event && event.source !== 'customer' && playSound( '/calypso/audio/chat-pling.wav' );
+};
+
+/**
+ * Action Handlers
+ */
+
+// Initialized this way for performance reasons
+export const handlers = Object.create( null );
+handlers[ HAPPYCHAT_RECEIVE_EVENT ] = onHappyChatMessage;
```

Comment:
non-blocking comment on naming here: does `onHappyChatMessage` tell us much? we generally want to avoid `onXYZ` names anyway. but what is the purpose of this function?

```js
handlers[ HAPPYCHAT_RECEIVE_EVENT ] = playDingOnCustomerMessages;
```

??